src/v2/ion/uiSchemaLabelTransformer.js

```diff
@@ -0,0 +1,142 @@
+  'select-authenticator-authenticate.authenticator.security_key': 'oie.webauthn',
+  'select-authenticator-authenticate.authenticator.security_question': 'oie.security.question',
+
+  'challenge-authenticator.email.credentials.passcode': 'mfa.challenge.enterCode.placeholder',
+  'challenge-authenticator.password.credentials.passcode': 'mfa.challenge.password.placeholder',
+  'challenge-authenticator.phone.credentials.passcode': 'mfa.challenge.enterCode.placeholder',
+  'challenge-authenticator.security_question.credentials.answer': 'mfa.challenge.answer.placeholder',
+};
+
+const getI18nKey = (i18nPath) => {
+  let i18nKey;
+  if (i18nPath.indexOf(SECURITY_QUESTION_PREFIX) === 0) {
+    const securityQuestionValue = i18nPath.replace(SECURITY_QUESTION_PREFIX, '');
+    i18nKey = `security.${securityQuestionValue}`;
+  }
+
+  if (I18N_OVERRIDE_MAPPINGS[i18nPath]) {
+    i18nKey = I18N_OVERRIDE_MAPPINGS[i18nPath];
+  }
+
+  if (i18nKey && !Bundles.login[i18nKey]) {
+    Logger.warn(`expect i18n key ${i18nKey} for ${i18nPath} but not found in 'login' bundle.`);
+  }
+};
+
+const overrideLabelIfI18NPathExists = (i18nPath, value) => {
+  const i18nKey = getI18nKey(i18nPath);
+  if (i18nKey) {
+    return loc(i18nKey, 'login');
+  } else {
+    return value;
+  }
+};
+
+const updateLabelForUiSchema = (remediation, uiSchema) => {
+  if (uiSchema.mutable === false) {
+    return;
+  }
+  Logger.info('i18n label transformer');
+  Logger.info('\t remediationName: ', remediation.name);
+  Logger.info('\t uiSchema: ', uiSchema);
+
+  const authenticatorType = remediation.relatesTo && remediation.relatesTo.value.type
+    ? `.${remediation.relatesTo.value.type}`
+    : '';
+  const i18nPrefix = `${remediation.name}${authenticatorType}.`;
+  const i18nPath = `${i18nPrefix}${uiSchema.name}`;
+
+  if (uiSchema.type === 'checkbox' && uiSchema.placeholder) {
+    Logger.info('\t 1: ', i18nPath);
+    uiSchema.placeholder = overrideLabelIfI18NPathExists(i18nPath, uiSchema.placeholder);
+  }
+
+  if (uiSchema.label) {
+    Logger.info('\t 2: ', i18nPath);
+    uiSchema.label = overrideLabelIfI18NPathExists(i18nPath, uiSchema.label);
+  }
+  if ($.isPlainObject(uiSchema.options)) {
+    uiSchema.options = _.mapObject(uiSchema.options, (value, key) => {
+      const i18nPathOption = `${i18nPath}.${key}`;
+      Logger.info('\t 3: ', i18nPathOption);
+      return overrideLabelIfI18NPathExists(i18nPathOption, value);
+    });
+  }
+  if (Array.isArray(uiSchema.options)) {
+    uiSchema.options.forEach(o => {
+      if (!o.label) {
+        return;
+      }
+      let i18nPathOption;
+      if (o.authenticatorType) {
+        i18nPathOption = `${i18nPath}.${o.authenticatorType}`;
+      } else if (o.value !== undefined) { // value could be string or number or undefined.
+        i18nPathOption = `${i18nPath}.${o.value}`;
+      }
+      Logger.info('\t 4: ', i18nPathOption);
+      overrideLabelIfI18NPathExists(i18nPathOption, o.label);
```

Comment:
`nit:` Do we need such a long name - `overrideLabelIfI18NPathExists`, how about something simpler - `overrideI18NLabel`. Whether if it exists or not can be left out to implementation details of the function.

Reply:
My thinking is that we can override only if it exists...

Reply:
> My thinking is that we can override only if it exists...

but the code will be slight different.

```
overrideI18NLabel = (uiSchema, key, i18nPath) => {
  if getI18nValue(i18nPath) then
    uiSchema[key] = getI18nValue(i18nPath);
}
...
overrideI18NLabel(uiSchema, 'label', i18nPath);

```

Looks like implementation is simpler.
I kind of leaning towards explicit function name at this case: `overrideLabelIfI18NKeyExists`